internal/config/sshSetup.go

```diff
@@ -0,0 +1,173 @@
+package config
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+)
+
+func createKeyPair(privateKeyPath, publicKeyPath string) error {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	// Generate and write private key as PEM
+	privateKeyFile, err := os.Create(privateKeyPath)
+	if err != nil {
+		return err
+	}
+	defer privateKeyFile.Close()
+
+	privateKeyPEM := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(privateKey),
+	}
+
+	if err := pem.Encode(privateKeyFile, privateKeyPEM); err != nil {
+		return err
+	}
+
+	// Set permissions to private key
+	if err = os.Chmod(privateKeyPath, 0400); err != nil {
+		return err
+	}
+
+	// Generate and write public key
+	pub, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(publicKeyPath, ssh.MarshalAuthorizedKey(pub), 0400)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func knownHost(host string, port int) bool {
+	var output []byte
+	var err error
+
+	if port != 22 {
+		output, err = exec.Command("ssh-keygen", "-H", "-F", fmt.Sprintf("%s:%d", host, port)).CombinedOutput()
+	} else {
+		output, err = exec.Command("ssh-keygen", "-H", "-F", host).CombinedOutput()
+	}
+
+	if err.Error() == "exit status 1" && len(output) > 0 {
+		return true
+	}
+
+	return false
+}
+
+func transferKey(keyPath, username, host string, port int) error {
+	command := exec.Command(
+		"ssh-copy-id",
+		"-i",
+		keyPath,
+		fmt.Sprintf("%s@%s", username, host),
+		"-p",
+		fmt.Sprintf("%d", port),
```

Comment:
I got another error, but I think this was related to my config (they were too many ssh keys I think). This fixed it.

```suggestion
		fmt.Sprintf("%d", port),
		"-o IdentitiesOnly=yes",
```